### PR TITLE
MNT: build wheels on GitHub Actions

### PR DESCRIPTION
**Pull request by kmuehlbauer**

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@djhoese Not sure why `freetype` isn't found correctly in the windows builds. Any idea?

**Comment by djhoese** [comment]
My browser (or github) doesn't really like the log for that job. It says it has 28k lines of output. I would have assumed any issue with freetype would have been resolved with that environment variable you set. Is there a chance the env var isn't getting down to the environment that is installing freetype-py?

**Comment by kmuehlbauer** [comment]
Yes, the pip install output and the npm stuff is really adding up. I'll try some more ideas I have tomorrow. Your guess might be true. We'll see.

Anyway, do we need the 32bit windows builds?

**Comment by djhoese** [comment]
I think since freetype-py doesn't provide 32-bit windows builds anymore we can't provide them (or rather it doesn't make a ton of sense to do it), but maybe I'm wrong. Windows 64-bit though we do need (should at least have wheels for the big 3 OSes).

**Comment by kmuehlbauer** [comment]
- win32 builds seem to not find `freetype.dll` in the testing stage

I've disabled win32 builds to have a look in the win64
- win64 builds do have problems with a specific Font (as it looks like)

I've tried several things so far but (changing cibuildwheel version, invoking cibuildwheel, adding MSBuild) to no avail...

So, if this can't be resolved easily, I would suggest to disable windows wheels for now on GitHub Actions (we can keep the azure pipelines workflow until resolving the issue).

**Comment by kmuehlbauer** [comment]
@rougier If you are around, could you have a look into the GitHub Actions part here? We have an issue with freetype on windows. Thanks a bunch!

**Comment by rougier** [comment]
Do you have  pointer to the log?

**Comment by kmuehlbauer** [comment]
https://github.com/vispy/vispy/pull/1974/checks?check_run_id=1740368581

**Comment by kmuehlbauer** [comment]
@rougier You might have to download the log to examine it. The GHA representation might be flaky somehow...

**Comment by kmuehlbauer** [comment]
Here's one of the failing tests for reference:

```
2021-01-21T08:14:07.0972711Z ================================== FAILURES ===================================
2021-01-21T08:14:07.0973199Z ______________ test_font_glyph[Bahnschrift SemiBold SemiConden] _______________
2021-01-21T08:14:07.0973569Z 
2021-01-21T08:14:07.0973806Z face = 'Bahnschrift SemiBold SemiConden'
2021-01-21T08:14:07.0973965Z 
2021-01-21T08:14:07.0974422Z     @pytest.mark.parametrize('face', ['OpenSans'] + sorted(sys_fonts))
2021-01-21T08:14:07.0974750Z     def test_font_glyph(face):
2021-01-21T08:14:07.0975000Z         """Test loading glyphs"""
2021-01-21T08:14:07.0975235Z         if face in known_bad_fonts:
2021-01-21T08:14:07.0976209Z             pytest.xfail()
2021-01-21T08:14:07.0976553Z         font_dict = dict(face=face, size=12, bold=False, italic=False)
2021-01-21T08:14:07.0976823Z         glyphs_dict = dict()
2021-01-21T08:14:07.0977077Z         chars = 'foobar^C&#'
2021-01-21T08:14:07.0977288Z         for char in chars:
2021-01-21T08:14:07.0977549Z             # Warning that Arial might not exist
2021-01-21T08:14:07.0977834Z             with warnings.catch_warnings(record=True):
2021-01-21T08:14:07.0978144Z                 warnings.simplefilter('always')
2021-01-21T08:14:07.0978432Z >               _load_glyph(font_dict, char, glyphs_dict)
2021-01-21T08:14:07.0978605Z 
2021-01-21T08:14:07.0978829Z vispy\util\fonts\tests\test_font.py:40: 
2021-01-21T08:14:07.0979280Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2021-01-21T08:14:07.0979752Z vispy\util\fonts\_freetype.py:50: in _load_glyph
2021-01-21T08:14:07.0980077Z     face = _load_font(f['face'], f['bold'], f['italic'])
2021-01-21T08:14:07.0980568Z vispy\util\fonts\_freetype.py:36: in _load_font
2021-01-21T08:14:07.0980867Z     fname = find_font(face, bold, italic)
2021-01-21T08:14:07.0981335Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2021-01-21T08:14:07.0981504Z 
2021-01-21T08:14:07.0981799Z face = 'Bahnschrift SemiBold SemiConden', bold = False, italic = False
2021-01-21T08:14:07.0982124Z orig_face = None
2021-01-21T08:14:07.0982260Z 
2021-01-21T08:14:07.0982528Z     def find_font(face, bold, italic, orig_face=None):
2021-01-21T08:14:07.0983032Z         style_dict = {'Regular': 0, 'Bold': 1, 'Italic': 2, 'Bold Italic': 3}
2021-01-21T08:14:07.0983300Z     
2021-01-21T08:14:07.0983593Z         # Figure out which font to actually use by trying to instantiate by name
2021-01-21T08:14:07.0984092Z         dc = user32.GetDC(0)  # noqa, analysis:ignore
2021-01-21T08:14:07.0984750Z         gdi32.SetGraphicsMode(dc, GM_ADVANCED)  # only TT and OT fonts
2021-01-21T08:14:07.0985027Z         logfont = LOGFONT()
2021-01-21T08:14:07.0985303Z         logfont.lfHeight = -12  # conv point to pixels
2021-01-21T08:14:07.0985598Z         logfont.lfWeight = FW_BOLD if bold else FW_NORMAL
2021-01-21T08:14:07.0985879Z         logfont.lfItalic = italic
2021-01-21T08:14:07.0988988Z         logfont.lfFaceName = face  # logfont needs Unicode
2021-01-21T08:14:07.0989584Z         hfont = gdi32.CreateFontIndirectW(byref(logfont))
2021-01-21T08:14:07.0990094Z         original = gdi32.SelectObject(dc, hfont)
2021-01-21T08:14:07.0990607Z         n_byte = gdi32.GetOutlineTextMetricsW(dc, 0, None)
2021-01-21T08:14:07.0990890Z         assert n_byte > 0
2021-01-21T08:14:07.0991308Z         metrics = OUTLINETEXTMETRIC()
2021-01-21T08:14:07.0991591Z         assert sizeof(metrics) >= n_byte
2021-01-21T08:14:07.0991918Z         assert gdi32.GetOutlineTextMetricsW(dc, n_byte, byref(metrics))
2021-01-21T08:14:07.0992440Z         gdi32.SelectObject(dc, original)
2021-01-21T08:14:07.0992862Z         user32.ReleaseDC(None, dc)
2021-01-21T08:14:07.0993389Z         use_face = cast(byref(metrics, metrics.otmpFamilyName), c_wchar_p).value
2021-01-21T08:14:07.0993704Z         if use_face != face:
2021-01-21T08:14:07.0994930Z             warnings.warn('Could not find face match "%s", falling back to "%s"'
2021-01-21T08:14:07.0995279Z                           % (orig_face or face, use_face))
2021-01-21T08:14:07.0995629Z         use_style = cast(byref(metrics, metrics.otmpStyleName), c_wchar_p).value
2021-01-21T08:14:07.0996002Z         use_style = style_dict.get(use_style, 'Regular')
2021-01-21T08:14:07.0996583Z         # AK: I get "Standaard" for use_style, which is Dutch for standard/regular
2021-01-21T08:14:07.0996989Z     
2021-01-21T08:14:07.0997430Z         # Now we match by creating private font collections until we find
2021-01-21T08:14:07.0997725Z         # the one that was used
2021-01-21T08:14:07.0997978Z         font_dir = op.join(os.environ['WINDIR'], 'Fonts')
2021-01-21T08:14:07.0998493Z         reg = winreg.ConnectRegistry(None, winreg.HKEY_LOCAL_MACHINE)
2021-01-21T08:14:07.0999982Z         key = 'SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Fonts'
2021-01-21T08:14:07.1000499Z         reg_vals = winreg.OpenKey(reg, key)
2021-01-21T08:14:07.1000833Z         n_values = winreg.QueryInfoKey(reg_vals)[1]
2021-01-21T08:14:07.1001106Z         fname = None
2021-01-21T08:14:07.1001355Z         for vi in range(n_values):
2021-01-21T08:14:07.1001648Z             name, ff = winreg.EnumValue(reg_vals, vi)[:2]
2021-01-21T08:14:07.1001969Z             if name.endswith('(TrueType)'):
2021-01-21T08:14:07.1002469Z                 ff = op.join(font_dir, ff) if op.basename(ff) == ff else ff
2021-01-21T08:14:07.1002803Z                 assert op.isfile(ff)
2021-01-21T08:14:07.1003804Z                 pc = c_void_p()
2021-01-21T08:14:07.1004145Z                 assert gdiplus.GdipNewPrivateFontCollection(byref(pc)) == 0
2021-01-21T08:14:07.1004564Z                 gdiplus.GdipPrivateAddFontFile(pc, ff)
2021-01-21T08:14:07.1004863Z                 family = c_void_p()
2021-01-21T08:14:07.1005209Z                 if gdiplus.GdipCreateFontFamilyFromName(use_face, pc,
2021-01-21T08:14:07.1005560Z                                                         byref(family)) == 0:
2021-01-21T08:14:07.1005817Z                     val = BOOL()
2021-01-21T08:14:07.1006123Z                     assert gdiplus.GdipIsStyleAvailable(family, use_style,
2021-01-21T08:14:07.1006777Z                                                         byref(val)) == 0
2021-01-21T08:14:07.1007006Z                     if val.value:
2021-01-21T08:14:07.1007259Z                         buf = create_unicode_buffer(LF_FACESIZE)
2021-01-21T08:14:07.1007610Z                         assert gdiplus.GdipGetFamilyName(family, buf, 0) == 0
2021-01-21T08:14:07.1007920Z >                       assert buf.value == use_face
2021-01-21T08:14:07.1008170Z E                       AssertionError
2021-01-21T08:14:07.1008316Z 
2021-01-21T08:14:07.1008547Z vispy\util\fonts\_win32.py:79: AssertionError
```

**Comment by rougier** [comment]
From what I read it fails to find some font (Bahnschrift SemiBold SemiConden). Is that a regular (system) font on windows?
Also the script that fails is named `_win32.py`, does that mean it test with win2 build ? Because as mentioned above, win32 build has been deprecated in freetype.

**Comment by kmuehlbauer** [comment]
@rougier Thanks for looking into this. The CI runs on GitHub Actions on their "windows-2019" runner. Bahnschrift is a semi-new font which is provided by microsoft (https://docs.microsoft.com/en-us/typography/font-list/bahnschrift). 

Hmm, so we use `find_font` which is included in `_freetype.py` from `_win32.py`. (All files below https://github.com/vispy/vispy/tree/master/vispy/util/fonts). Would we need to adapt this to work with 64bit windows?

**Comment by kmuehlbauer** [comment]
The weird thing is, that the wheelbuilding runs on azure pipelines for both win32 and win64 without failures.

**Comment by kmuehlbauer** [comment]
@rougier Regarding Bahnschift maybe related: https://stackoverflow.com/questions/57111097/does-freetype-support-variable-fonts Any thoughts on that?

**Comment by rougier** [comment]
Oh, it this is a variable font, that might be very well the problem. I think freetype started support but I don't think it is fully operational not mentioning freetype-py. Variable font means you have a single file that can interpolate between different features. Typically for eights, you define 2 or three weights in the file and for other weights they an be interpolated. Not sure it is really needed for vispy.

**Comment by kmuehlbauer** [comment]
@rougier What do you suggest to resolve this? Can these fonts be disabled somehow, or can we just make it work with a subset of that font?

**Comment by rougier** [comment]
I've no idea where these fonts are used on Vispy and/or why are they tested. I think we can stick to have a sane default font stack and test only that. From [https://github.com/vispy/vispy/blob/master/vispy/util/fonts/tests/test_font.py](test_font.py), I think the problem is that all the system fonts are tested. Since we cannot render some fonts, we shoudl just discard the font and say the test is ok.

**Comment by kmuehlbauer** [comment]
@rougier Thanks for coming back. I'll try to fix the tests accordingly.

**Comment by kmuehlbauer** [comment]
@rougier I'll start with discarding the failing Font's for now.

**Comment by kmuehlbauer** [comment]
@larsoner Do you have any idea if the `_win32.py` (https://github.com/vispy/vispy/pull/1974#issuecomment-768174809) is for bot win32 and win64 or if we need a `_win64` too.

**Comment by kmuehlbauer** [comment]
Seems that the Bahnschrift issue is resolved...

**Comment by kmuehlbauer** [comment]
Something fishy with Azure here, close/reopen didn't help. Just trying with an empty commit now.

**Comment by kmuehlbauer** [comment]
@djhoese This is ready from my side.

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # if: github.event_name == 'release' && github.event.action == 'published'
+    steps:
+    - uses: actions/download-artifact@v2
+      with:
+        name: artifact
+        path: dist
+    - uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.VISPY_PYPI_TOKEN }}
```
Comment (larsoner) [inline .github/workflows/wheels.yml:81]: Probably needs to be added to 

https://github.com/vispy/vispy/settings/secrets/actions

Comment (kmuehlbauer) [reply .github/workflows/wheels.yml:81]: Yes, correct, that would need to be added. Thanks for the reminder. I've talked about that with @djhoese the other day on gitter. Good chance that we would have found out first when things brake.

Comment (larsoner) [reply .github/workflows/wheels.yml:81]: Okay I did it (I think)

**Review by larsoner — APPROVED** [review]
LGTM +1 for merge.

**Review by djhoese — CHANGES_REQUESTED** [review]
I haven't read all of the changes but I think you could remove the Azure config so we don't have two systems building wheels.

**Review by djhoese — APPROVED** [review]
Everything looks good, but remove the azure config and feel free to merge. After merge I'll make sure to disable the Azure project.

**Comment by kmuehlbauer** [comment]
@djhoese Azure config has been removed.

**Comment by djhoese** [comment]
Thanks!

**Pull request merged**